novem/table/selector.py

```diff
@@ -18,6 +18,46 @@ def __init__(self, message: str):
         super().__init__(message)
 
 
+def handle_position(position: Union[slice, Any, int], used_positions: Set[int]) -> List[int]:
+    """
+    Handle different types of position results from get_loc()
+
+    Args:
+        position: Result from get_loc() - can be int, slice, or boolean array
+        used_positions: Set of already used positions to avoid duplicates
+
+    Returns:
+        List of integer positions
+    """
+    if isinstance(position, slice):
+        # Convert slice to list of positions
+        start = position.start if position.start is not None else 0
+        stop = position.stop if position.stop is not None else float("inf")
+        step = position.step if position.step is not None else 1
+        positions = list(range(start, stop, step))
+        # Only take first position that hasn't been used
+        for pos in positions:
+            if pos not in used_positions:
+                used_positions.add(pos)
+                return [pos]
+        return []
+    elif isinstance(position, np.ndarray):  # type: ignore
+        # Handle boolean mask or integer array
+        if position.dtype == bool:
+            position = np.flatnonzero(position)
+        position = [pos for pos in position if pos not in used_positions]
+        if position:
+            used_positions.add(position[0])
+            return [position[0]]
```

Comment:
```suggestion
            return position[0:1]
```
is maybe a bit nicer

Reply:
Nice suggestion! I think i prefer as is for this particular function (as the surrounding return values are all explicit arrays wrapped in `[]`). But will definitely keep this in mind in the future